src/data-types/datetime2.ts

```diff
@@ -53,8 +56,11 @@ module.exports = {
       } else {
         timestamp = ((time.getHours() * 60 + time.getMinutes()) * 60 + time.getSeconds()) * 1000 + time.getMilliseconds();
       }
-
-      timestamp = timestamp * Math.pow(10, parameter.scale - 3);
+      if(parameter.scale){
+        timestamp = timestamp * Math.pow(10, parameter.scale - 3);
```

Comment:
In cases like this, where a value like `parameter.scale` is optional based on it's type definition but we know it's definately defined at this point in the code, you can add a `!` at the end to hint to TypeScript that the value is not null or undefined:

```suggestion
    timestamp = timestamp * Math.pow(10, parameter.scale! - 3);
```

That'll allow you to remove the `if`...`else` statement again.